### PR TITLE
spec: update notation cli md file as index for sub-commands

### PR DESCRIPTION
**Pull request by yizha1**
Update notation cli md file to reflect the changes in rc.1 release and add references to each sub-command.

### CODE REVIEW

File: specs/notation-cli.md [hunk 1]
```diff
@@ -1,284 +1,44 @@
...
+Available Commands:
+  certificate Manage certificates in trust store
+  help        Help about any command
+  key         Manage keys used for signing
+  list        List signatures of the signed artifact
+  login       Login to registries
+  logout      Log out from the logged in registries
+  plugin      Manage plugins
+  sign        Sign artifacts
+  verify      Verify artifacts
```
Comment (priteshbandi) [inline specs/notation-cli.md:36]:
```suggestion
  sign        Signs artifacts
  verify      Verifies artifacts signature
```

File: specs/notation-cli.md [hunk 2]
```diff
@@ -1,284 +1,44 @@
...
+  version     Print the version
```
Comment (priteshbandi) [inline specs/notation-cli.md:37]: Prints the version of notation CLI.

File: specs/notation-cli.md [hunk 3]
```diff
@@ -1,284 +1,44 @@
...
+
+```text
+Notation - Notary V2 - a tool to sign, store, and verify artifacts
+
+Usage:
+  notation [command]
+
+Available Commands:
+  certificate Manage certificates in trust store
+  help        Help about any command
```
Comment (priteshbandi) [inline specs/notation-cli.md:29]:
```suggestion
  help        Provides a terse summary of all options.
```

File: specs/notation-cli.md (hunk 1 above)
Comment (yizha1) [reply specs/notation-cli.md:36]: Why do we need to use singular?  We didn't use singular for other specs as well.

Comment (priteshbandi) [reply specs/notation-cli.md:36]: We can ignore singular part of comment.

**Review by priteshbandi — APPROVED** [review]
LGTM

File: specs/notation-cli.md (hunk 3 above)
Comment (yizha1) [reply specs/notation-cli.md]: Use "Provide a terse summary of all options" instead.

File: specs/notation-cli.md [hunk 4]
```diff
@@ -1,284 +1,44 @@
...
+| ------------------------------------------- | -------------------------------------- |
+| [certificate](./commandline/certificate.md) | Manage certificates in trust store     |
+| [key](./commandline/key.md)                 | Manage keys used for signing           |
+| [list](./commandline/list.md)               | List signatures of the signed artifact |
+| [login](./commandline/login.md)             | Login to registries                    |
+| [logout](./commandline/logout.md)           | Log out from the logged in registries  |
+| [plugin](./commandline/plugin.md)           | Manage plugins                         |
+| [sign](./commandline/sign.md)               | Sign artifacts                         |
+| [verify](./commandline/verify.md)           | Verify artifacts                       |
+| [version](./commandline/version.md)         | Print the version of notation CLI      |
```
Comment (patrickzheng200) [inline specs/notation-cli.md:17]: Do we need to add [help] in the Command list?

**Review by dtzar — APPROVED** [review]

**Review by vaninrao10 — CHANGES_REQUESTED** [review]
1.	Change the first paragraph from alpha 1 to alpha 4 verbage. 
2.	If we are eliminating cache concept, then do we need cache command in this spec?

**Comment by yizha1** [comment]
> 1. Change the first paragraph from alpha 1 to alpha 4 verbage.
> 2. If we are eliminating cache concept, then do we need cache command in this spec?

@vaninrao10  maybe you saw the wrong file. The version info and cache command are already removed in the new spec

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: specs/notation-cli.md [hunk 5]
```diff
@@ -1,284 +1,45 @@
...
+
+## Notation Outline
+
+```text
+Notation - Notary V2 - a tool to sign, store, and verify artifacts
```
Comment (rgnote) [inline specs/notation-cli.md:23]:
```suggestion
Notation - Notary V2 - a tool to sign and verify OCI artifacts
```

File: specs/notation-cli.md [hunk 6]
```diff
@@ -1,284 +1,45 @@
...
+  login       Login to registries
+  logout      Log out from the logged in registries
+  plugin      Manage plugins
+  sign        Sign artifacts
+  verify      Verify artifacts
+  version     Print the version of notation CLI
+
+Flags:
+  -h, --help         help for notation
+      --plain-http   Registry access via plain HTTP
```
Comment (rgnote) [inline specs/notation-cli.md:42]:
```suggestion
  -h, --help        Help for notation
  --plain-http     Registry access via plain HTTP
```
```suggestion
  -h, --help         Help for notation
  --plain-http      Registry access via plain HTTP
```

File: specs/notation-cli.md [hunk 7]
```diff
@@ -1,284 +1,45 @@
...
+
+```text
+Notation - Notary V2 - a tool to sign, store, and verify artifacts
+
+Usage:
+  notation [command]
+
+Available Commands:
+  certificate Manage certificates in trust store
+  help        Provide a terse summary of all options
```
Comment (rgnote) [inline specs/notation-cli.md:30]: This looks like we have a `help` command but in the above table, `help` points to `#notation-outline`. It doesn't make sense to have have `notation help --help` option.

File: specs/notation-cli.md (hunk 6 above)
Comment (yizha1) [reply specs/notation-cli.md:42]: `--plain-http` should be aligned with `--help`

File: specs/notation-cli.md (hunk 4 above)
Comment (yizha1) [reply specs/notation-cli.md:17]: No, it will be removed

**Pull request merged**